e2e/template_e2e_tests/run_e2e_test.sh

```diff
@@ -133,63 +132,74 @@ test_api_endpoints_cloudrun() {
   CLOUDRUN_PYTEST_STATUS=${PIPESTATUS[0]}
 }
 
+# Check API tests result
+print_api_test_result() {
+  if [[ $GKE_PYTEST_STATUS == 0 && $CLOUDRUN_PYTEST_STATUS == 0 ]]; then
+    echo "API Tests passed."
+  else
+    echo -e '\033[31m ERROR: API Tests failed \033[0m'
+    echo "GKE_PYTEST_STATUS=$GKE_PYTEST_STATUS"
+    echo "CLOUDRUN_PYTEST_STATUS=$CLOUDRUN_PYTEST_STATUS"
+    exit -1
+  fi
+}
+
 # Clean up GCP resources.
 clean_up() {
-  cd $BASE_DIR/terraform/stages/gke
-  terraform destroy -auto-approve
-  
-  cd $BASE_DIR/terraform/stages/cloudrun
-  terraform destroy -auto-approve
-  
-  cd $BASE_DIR/terraform/stages/foundation
-  terraform destroy -auto-approve
+  if [[ "$skip_cleanup" ==  "" ]]; then
+    printf "Cleaning up ${PROJECT_ID}...\n"
+    
+    echo "BASE_DIR=$BASE_DIR"
+    export TF_BUCKET_NAME="${PROJECT_ID}-tfstate"
+    
+    cd $BASE_DIR/terraform/stages/gke
+    terraform init -reconfigure -backend-config=bucket=$TF_BUCKET_NAME
```

Comment:
The bucket name can be provided in `backend.tf`

Reply:
The bucket name needs to be variable in the template.

Reply:
That is correct

Reply:
Terraform backend doesn't allow variable, so using -backend-config is the only way: https://developer.hashicorp.com/terraform/language/settings/backends/configuration#using-a-backend-block